pubsub/azure/servicebus/message.go

```diff
@@ -138,6 +139,77 @@ func NewPubsubMessageFromASBMessage(asbMsg *azservicebus.ReceivedMessage, topic
 	return pubsubMsg, nil
 }
 
+func NewBulkMessageEntryFromASBMessage(asbMsg *azservicebus.ReceivedMessage) (pubsub.BulkMessageEntry, error) {
+	entryID, err := uuid.NewRandom()
+	if err != nil {
+		// This should never happen.
+		return pubsub.BulkMessageEntry{}, err
+	}
+
+	bulkMsgEntry := pubsub.BulkMessageEntry{
+		EntryID: entryID.String(),
+	}
+
+	bulkMsgEntry.Event = asbMsg.Body
+
+	addToMetadata := func(msg pubsub.BulkMessageEntry, key, value string) {
+		if msg.Metadata == nil {
+			msg.Metadata = make(map[string]string)
+		}
+
+		msg.Metadata[fmt.Sprintf("metadata.%s", key)] = value
```

Comment:
How about `msg.Metadata["metadata." + key] = value` ? Without going through all the work of a Sprintf

Reply:
Refactored this to move to a common implementation.